vms/evm/subscriber_test.go

```diff
@@ -0,0 +1,131 @@
+package evm
+
+import (
+	"math/big"
+	"os"
+	"testing"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/awm-relayer/config"
+	"github.com/ava-labs/awm-relayer/database"
+	mock_ethclient "github.com/ava-labs/awm-relayer/vms/evm/mocks"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ava-labs/subnet-evm/interfaces"
+	"github.com/ava-labs/subnet-evm/x/warp"
+	"github.com/ethereum/go-ethereum/common"
+	"go.uber.org/mock/gomock"
+)
+
+func makeSubscriberWithMockEthClient(t *testing.T) (*subscriber, *mock_ethclient.MockClient) {
+	sourceSubnet := config.SourceSubnet{
+		SubnetID:          "2TGBXcnwx5PqiXWiqxAKUaNSqDguXNh1mxnp82jui68hxJSZAx",
+		ChainID:           "S4mMqUXe7vHsGiRAma6bv3CKnyaLssyAxmQ2KvFpX1KEvfFCD",
+		VM:                config.EVM.String(),
+		APINodeHost:       "127.0.0.1",
+		APINodePort:       9650,
+		EncryptConnection: false,
+		RPCEndpoint:       "https://subnets.avax.network/mysubnet/rpc",
+	}
+
+	logger := logging.NewLogger(
+		"awm-relayer-test",
+		logging.NewWrappedCore(
+			logging.Info,
+			os.Stdout,
+			logging.JSON.ConsoleEncoder(),
+		),
+	)
+
+	subnetId, err := ids.FromString(sourceSubnet.ChainID)
+	if err != nil {
```

Comment:
lets use `require.NoError` with error message instead of `t.Fatalf`

Reply:
thanks, addressed in https://github.com/ava-labs/awm-relayer/pull/89/commits/75c965b79d0cb866b61ac1757909f7da700ab936